src/StakeRegistry.sol

```diff
@@ -503,18 +547,27 @@ contract StakeRegistry is StakeRegistryStorage {
         uint96 weight;
         uint256 stratsLength = strategyParamsLength(quorumNumber);
         StrategyParams memory strategyAndMultiplier;
-
-        address[] memory operators = new address[](1);
-        operators[0] = operator;
-        uint32 beforeTimestamp = uint32(block.timestamp + slashableStakeLookAhead);
-        (uint256[][] memory strategyShares, ) = IAllocationManager(serviceManager.allocationManager()).getMinDelegatedAndSlashableOperatorShares(OperatorSet(address(serviceManager), quorumNumber), operators ,strategiesPerQuorum[quorumNumber], beforeTimestamp);
-        for (uint256 i = 0; i < stratsLength; i++) {
-            // accessing i^th StrategyParams struct for the quorumNumber
-            strategyAndMultiplier = strategyParams[quorumNumber][i];
-
-            // add the weight from the shares for this strategy to the total weight
-            if (strategyShares[i][0] > 0) {
-                weight += uint96(strategyShares[i][0] * strategyAndMultiplier.multiplier / WEIGHTING_DIVISOR);
+        uint256[] memory strategyShares;
+
+        if (stakeTypePerQuorum[quorumNumber]== StakeType.TOTAL_SLASHABLE) {
+            strategyShares = _getSlashableStakePerStrategy(quorumNumber, operator);
+            for (uint256 i = 0; i < stratsLength; i++) {
+                strategyAndMultiplier = strategyParams[quorumNumber][i];
+                if (strategyShares[i] > 0) {
+                    weight += uint96(strategyShares[i] * strategyAndMultiplier.multiplier / WEIGHTING_DIVISOR);
+                }
+            }
+        } else {
+            /// M2 Concept of delegated stake
+            strategyShares = delegation.getOperatorShares(operator, strategiesPerQuorum[quorumNumber]);
+            for (uint256 i = 0; i < stratsLength; i++) {
```

Comment:
duplicated loop move outside conditional

Reply:
can you provide a bit more color, it's not clear what you mean

Reply:
basically do:

```solidity
for (uint256 i = 0; I < stratsLength; i++) {
       If slashable, get strategyShares one way. Else Get the other way. 
       
       weight +=...
}
```

Reply:
ahh gotcha

Reply:
all set